### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by RReverser**

### CODE REVIEW

File: src/vnode.js [hunk 1]
```diff
@@ -2,8 +2,9 @@
  * Virtual DOM Node
  */
 export function VNode(nodeName, opt_attributes, opt_children) {
+	opt_attributes=opt_attributes||{};
```
Comment (developit) [inline src/vnode.js:5]: I think I might move this into `src/h.js` so the args to `VNode` are always the same type

File: src/dom/index.js [hunk 2]
```diff
@@ -70,13 +70,17 @@ export function setAccessor(node, name, old, value, isSvg) {
 	}
 	else {
 		let ns = isSvg && name.match(/^xlink\:?(.+)/);
+		if (ns) {
+			name = toLowerCase(ns[1]);
+			ns = 'http://www.w3.org/1999/xlink';
+		} else {
+			ns = '';
```
Comment (developit) [inline src/dom/index.js:77]: I ended up [benchmarking this](https://esbench.com/bench/58c4cc3499634800a0347a12) and it seems like `undefined` performs better than `''`:
screen shot 2017-03-12 at 11 41 54 am

File: externs.js [hunk 3]
```diff
@@ -45,21 +45,6 @@ Component.prototype.prevContext;
 Component.prototype.base;
 /** @type {?} */
 Component.prototype.nextBase;
-/** @type {?} */
-Component.prototype._parentComponent;
-/** @type {?} */
-Component.prototype._component;
```
Comment (developit) [inline externs.js:51]: _might_ need to keep `_component` as this mirrors the dom annotations.

Comment (RReverser) [reply externs.js:51]: Not sure what you mean.

Comment (developit) [reply externs.js:51]: Preact leaves component references on Dom nodes using a _component property, same as it does compositional children.

Comment (RReverser) [reply externs.js:51]: So... is it a problem if the property name gets minified?

Comment (developit) [reply externs.js:51]: shouldn't be for most people since it's not publicized, but some tooling might rely on it (definitely the devtools integration)

**Pull request merged**